pkg/validation/app.go

```diff
@@ -428,6 +386,52 @@ func (v *Validator) validateUserConfig(ctx context.Context, cr v1alpha1.App) err
 	return nil
 }
 
+func (v *Validator) validateConfigMapExistence(ctx context.Context, name, namespace, kind string, cr v1alpha1.App) error {
+	if namespace == "" {
+		return microerror.Maskf(validationError, namespaceNotFoundReasonTemplate, kind, name)
+	}
+
+	if name == "" {
+		return microerror.Maskf(validationError, nameNotFoundReasonTemplate, kind)
+	}
+
+	if v.conditional && key.IsManagedByFlux(cr, v.projectName) {
```

Comment:
```suggestion
	if v.validateResourcesExist && key.IsManagedByFlux(cr, v.projectName) {
```

Naming is hard here but I think this is clearer. WDYT?